include/boost/config/detail/suffix.hpp

```diff
@@ -1198,6 +1198,14 @@ namespace std{ using ::type_info; }
 #endif
 #endif
 
+#ifdef __has_include
+#if !__has_include(<version>)
+#  define BOOST_NO_CXX20_HDR_VERSION
+#endif
```

Comment:
I think that it would be useful to have an `#else` here that does `#include <version>` so that we can rely on it being always included if available, even if `select_stdlib_config` didn't include it because of a user override.